s3proxy/internal/crypto/crypto.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+/*
+Package crypto provides encryption and decryption functions for the s3proxy.
```

Comment:
While this package's job seems simple, I'd suggest to use Tink's AEAD encryption: https://pkg.go.dev/github.com/google/tink/go/aead#example-package. This makes it even harder to make crypto mistakes.

Reply:
If we decide to still implement this ourselves, then we should at least add tests for this.

Reply:
Addressed here: https://github.com/edgelesssys/constellation/pull/2398